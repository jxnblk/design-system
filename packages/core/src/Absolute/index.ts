export { default as Absolute } from './Absolute'
