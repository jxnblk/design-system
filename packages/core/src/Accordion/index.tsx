/* istanbul ignore file */
// todo: remove coverage ignore once storybook interaction test coverage counts
export { Accordion } from './Accordion'
