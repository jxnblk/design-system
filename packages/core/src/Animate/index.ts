export { Animate, type AnimateProps } from './Animate'
