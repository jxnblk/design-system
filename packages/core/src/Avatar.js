import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Flex from './Flex'
import Heading from './Heading'
import Text from './Text'
import { User } from 'pcln-icons'

const StyledImage = styled(Flex)`
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-size: 40px;
  background-image: url(${props => props.src});
`

function Avatar({ className, title, subTitle, src, color }) {
  return (
    <Flex className={className}>
      <StyledImage src={src} color={color} p={2}>
        {!src && <User />}
      </StyledImage>
      {title && (
        <Flex flexDirection="column" justifyContent="center" ml={3}>
          <Heading.h5 m={0}>{title}</Heading.h5>
          <Text color="text.light" fontSize={0}>
            {subTitle}
          </Text>
        </Flex>
      )}
    </Flex>
  )
}

Avatar.displayName = 'Avatar'

Avatar.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  src: PropTypes.string,
  color: PropTypes.string
}

Avatar.defaultProps = {
  className: '',
  color: 'primary'
}

export default Avatar
