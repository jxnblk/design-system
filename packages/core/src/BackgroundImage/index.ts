export { BackgroundImage, type BackgroundImageProps } from './BackgroundImage'
