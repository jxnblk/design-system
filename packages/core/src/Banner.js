import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import CloseButton from './CloseButton'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'
import { applyVariations, hasPaletteColor, deprecatedColorValue } from './utils'
import {
  Attention as AttentionIcon,
  Information as InformationIcon,
  Success as SuccessIcon,
  Warning as WarningIcon
} from 'pcln-icons'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: <SuccessIcon />
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: <SuccessIcon />
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: <WarningIcon />
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: <WarningIcon />
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: <AttentionIcon />
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: <InformationIcon />
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: <InformationIcon />
  }
}

const StyledBox = styled(Box)`
  ${applyVariations('Banner')}
`

const Banner = props => {
  const bannerColor =
    bannerColors[
      !props.bg && props.color === 'green' ? props.color : props.bg
    ] || {}
  const Icon = props.icon || bannerColor.icon
  const color =
    hasPaletteColor(props) || !bannerColor.color
      ? props.color
      : bannerColor.color
  const header = React.isValidElement(props.header) ? (
    props.header
  ) : (
    <Text textStyle="display2">{props.header}</Text>
  )

  return (
    <StyledBox
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!Icon &&
          !!props.showIcon &&
          React.cloneElement(Icon, { mr: 2, size: 24, mt: '-2px' })}
        <Box textAlign={props.textAlign} width={1}>
          {props.header && header}
          <Text.span fontSize={1}>{props.text}</Text.span>
          {props.children}
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </StyledBox>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  icon: PropTypes.node,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string,
  color: deprecatedColorValue(),
  bg: deprecatedColorValue()
}

Banner.defaultProps = {
  textAlign: 'left',
  showIcon: true,
  color: 'green'
}

export default withTheme(Banner)
