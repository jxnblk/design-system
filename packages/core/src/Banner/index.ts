export { Banner, type BannerProps, type BannerColor } from './Banner'
