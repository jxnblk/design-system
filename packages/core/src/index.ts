import * as storybookArgs from './storybook/args'

export type { IButtonProps } from './Button'
export type { IDialogProps } from './Dialog'
export type { IGridProps } from './Grid'
export type { ICloseButtonProps } from './CloseButton'

export { Absolute } from './Absolute'
export { Accordion } from './Accordion'
export * from './Animate'
export { Avatar } from './Avatar'
export { BackgroundImage } from './BackgroundImage'
export { Badge } from './Badge'
export { Banner } from './Banner'
export { BlockLink } from './BlockLink'
export { Box } from './Box'
export { Breadcrumbs } from './Breadcrumbs'
export { Button } from './Button'
export { Card } from './Card'
export { Checkbox } from './Checkbox'
export { ButtonChip, FilterChip as Chip, ChoiceChip, FilterChip } from './Chip'
export { CloseButton } from './CloseButton'
export * from './Container'
export { Dialog } from './Dialog'
export { Divider } from './Divider'
export { DotLoader } from './DotLoader'
export { Flag } from './Flag'
export { Flex } from './Flex'
export { FormField, InputField } from './FormField'
export { GenericBanner } from './GenericBanner'
export * from './Grid'
export { Heading } from './Heading'
export { Hide } from './Hide'
export { Hug } from './Hug'
export { Icon } from './Icon'
export { IconButton } from './IconButton'
export { IconField } from './IconField'
export { Image } from './Image'
export { Input } from './Input'
export { InputGroup } from './InputGroup'
export { Label } from './Label'
export * from './Layout'
export { Link } from './Link'
export { List } from './List'
export { PasswordInput } from './PasswordInput'
export { PlaceholderImage } from './PlaceholderImage'
export * from './Popout'
export * from './ProgressBar'
export { Radio } from './Radio'
export { RadioCheckToggleCard } from './RadioCheckToggleCard'
export { RatingBadge } from './RatingBadge'
export { Relative } from './Relative'
export { Select } from './Select'
export { ShadowEffect, ShadowOverlay } from './ShadowEffect'
export { Shimmer } from './Shimmer'
export { SkipMenu } from './SkipMenu'
export { SlideBox } from './SlideBox'
export { Spinner } from './Spinner'
export { SrOnly } from './SrOnly'
export { Stamp } from './Stamp'
export { Stepper } from './Stepper'
export { Swatch } from './Swatch'
export { Text } from './Text'
export { TextArea } from './TextArea'
export { Toast } from './Toast'
export { ToastProvider } from './ToastProvider'
export { Toggle } from './Toggle'
export { ToggleBadge } from './ToggleBadge'
export { Tooltip } from './Tooltip'
export { Truncate } from './Truncate'

export { ThemeProvider } from './ThemeProvider'

export { useToast } from './ToastProvider'

export * from './utils'

// DocsUtils
export * from './DocsUtils/DoDont'
export * from './DocsUtils/DocTable'
export * from './DocsUtils/Hero'
// export * from './DocsUtils/LiveDemo'
export * from './DocsUtils/Note'
export * from './DocsUtils/RelatedComponent'
export * from './DocsUtils/Section'
export * from './DocsUtils/StoryHeading'
export * from './DocsUtils/StoryStage'
export * from './DocsUtils/TableOfContents'
export { storybookArgs }

export * from './theme'
