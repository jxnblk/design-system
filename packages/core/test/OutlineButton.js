import React from 'react'
import { OutlineButton, theme } from '../src'
import styled from 'styled-components'

describe('OutlineButton', () => {
  test('renders', () => {
    const json = rendererCreateWithTheme(<OutlineButton />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('disabled prop sets', () => {
    const json = rendererCreateWithTheme(<OutlineButton disabled />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('color', '#e8f2ff')
  })

  test('without disabled prop sets', () => {
    const json = rendererCreateWithTheme(<OutlineButton />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue, {
      modifier: ':hover'
    })
  })

  test('renders with nested style', () => {
    const Component = styled.div`
      ${OutlineButton} {
        color: red;
      }
    `
    const wrapper = rendererCreateWithTheme(
      <Component>
        <OutlineButton />
      </Component>
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
