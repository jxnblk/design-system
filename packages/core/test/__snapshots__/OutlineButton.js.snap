// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: default;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  color: #e8f2ff;
  box-shadow: inset 0 0 0 2px #e8f2ff;
  background-color: transparent;
}

.c0:hover {
  background-color: transparent;
}

<button
  className="c0"
  color="primary"
  disabled={true}
/>
`;

exports[`OutlineButton renders 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  color: #007aff;
  box-shadow: inset 0 0 0 2px #007aff;
  background-color: transparent;
}

.c0:hover {
  background-color: transparent;
  color: #049;
  box-shadow: inset 0 0 0 2px #049;
}

<button
  className="c0"
  color="primary"
/>
`;

exports[`OutlineButton renders with nested style 1`] = `
.c2 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  color: #007aff;
  box-shadow: inset 0 0 0 2px #007aff;
  background-color: transparent;
}

.c2:hover {
  background-color: transparent;
  color: #049;
  box-shadow: inset 0 0 0 2px #049;
}

.c0 .c1 {
  color: red;
}

<div
  className="c0"
>
  <button
    className="c1 c2"
    color="primary"
  />
</div>
`;

exports[`OutlineButton without disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  color: #007aff;
  box-shadow: inset 0 0 0 2px #007aff;
  background-color: transparent;
}

.c0:hover {
  background-color: transparent;
  color: #049;
  box-shadow: inset 0 0 0 2px #049;
}

<button
  className="c0"
  color="primary"
/>
`;
